### PR TITLE
Limit concurrent video upload count to 2

### PR DESCRIPTION
**Pull request by ua741**

### CODE REVIEW

File: lib/utils/file_uploader.dart [hunk 1]
```diff
@@ -229,17 +232,31 @@ class FileUploader {
...
+      var pendingEntry = _queue.entries
           .firstWhere(
             (entry) => entry.value.status == UploadStatus.not_started,
             orElse: () => null,
           )
           ?.value;
-      if (firstPendingEntry != null) {
-        firstPendingEntry.status = UploadStatus.in_progress;
+
+      if (pendingEntry == null &&
```
Comment (ua741) [inline lib/utils/file_uploader.dart:242]: not null

File: lib/utils/file_uploader.dart [hunk 2]
```diff
@@ -229,17 +232,31 @@ class FileUploader {
...
             (entry) => entry.value.status == UploadStatus.not_started,
             orElse: () => null,
           )
           ?.value;
-      if (firstPendingEntry != null) {
-        firstPendingEntry.status = UploadStatus.in_progress;
+
+      if (pendingEntry != null &&
+          pendingEntry.file.fileType == FileType.video &&
+          _currentlyVideoUploading < kMaximumConcurrentVideoUploads) {
```
Comment (vishnukvmd) [inline lib/utils/file_uploader.dart:244]: Flip the case?

File: lib/utils/file_uploader.dart [hunk 3]
```diff
@@ -50,6 +52,7 @@ class FileUploader {
   // Upload session is the period between the first entry into the _queue and last entry out of the _queue
   int _totalCountInUploadSession = 0;
   int _currentlyUploading = 0;
```
Comment (vishnukvmd) [inline lib/utils/file_uploader.dart:54]: Let's rename these to `_uploadCounter` and `_videoUploadCounter`?

File: lib/utils/file_uploader.dart [hunk 4]
```diff
@@ -22,6 +22,7 @@ import 'package:photos/events/subscription_purchased_event.dart';
 import 'package:photos/main.dart';
 import 'package:photos/models/encryption_result.dart';
 import 'package:photos/models/file.dart';
+import 'package:photos/models/file_type.dart';
```
[the remark this answers is not in the document]
Comment (ua741) [reply lib/utils/file_uploader.dart:25]: fileType will be needed to check if it's video?

File: lib/utils/file_uploader.dart [hunk 5]
```diff
@@ -228,18 +232,32 @@ class FileUploader {
...
             (entry) => entry.value.status == UploadStatus.not_started,
             orElse: () => null,
           )
           ?.value;
-      if (firstPendingEntry != null) {
-        firstPendingEntry.status = UploadStatus.in_progress;
+
+      if (pendingEntry != null &&
+          pendingEntry.file.fileType == FileType.video &&
+          _videoUploadCounter > kMaximumConcurrentVideoUploads) {
```
Comment (ua741) [inline lib/utils/file_uploader.dart:245]: equals to missing 😡

**Review by vishnukvmd — APPROVED** [review]

**Review by vishnukvmd — APPROVED** [review]

**Pull request merged**